src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -148,14 +149,19 @@ export const useDiscoverHistogram = ({
   useEffect(() => {
     const subscription = createTotalHitsObservable(unifiedHistogram?.state$)?.subscribe(
       ({ status, result }) => {
+        const { recordRawType, result: totalHitsResult } = savedSearchData$.totalHits$.getValue();
+
+        if (recordRawType === RecordRawType.PLAIN) {
+          // ignore histogram's total hits updates for text-based records as Discover manages them during docs fetching
+          return;
+        }
```

Comment:
TBH rather than having special logic for this case in Discover, I think it would probably be better to just remove the ES|QL total hits logic from `src/plugins/unified_histogram/public/chart/histogram.tsx` and skip calling `onTotalHitsChange` like we're doing when the chart is hidden. It's probably not necessary to have Unified Histogram update total hits when the chart is visible, but not when it's hidden, and instead just remove all the ES|QL total hits logic from Unified Histogram.

Reply:
Adding this check as a safeguard on Discover side seemed less risky than updating `src/plugins/unified_histogram/public/chart/histogram.tsx` which could affect data view mode.  I would leave this work for later when we continue refactoring for https://github.com/elastic/kibana/issues/177156 if that's okay with you.

Reply:
I don't think it would be too bad since the ES|QL part of the logic in there is pretty contained, but I'm ok with leaving this for later 👍 We can either address it separately as a followup or include it in a PR that addresses #177156 too.